src/pages/data-entry/data-entry.jsx

```diff
@@ -1,127 +1,214 @@
-import React, { Component } from "react";
+import React, { Component } from 'react'
+import { useQuery } from '@apollo/react-hooks'
+import { useMutation } from '@apollo/react-hooks'
+import { gql } from 'apollo-boost'
+
+const units = gql`
+  {
+    getAllAvailableUnits
+    }
+  }
+`
+
+const insertMutation = gql`
+  mutation InsertMutation(
+    $unit: String!
+    $value: Float!
+    $latitude: Float!
+    $longitude: Float!
+  ) {
+    insertValue(
+      unit: $unit
+      value: $value
+      latitude: $latitude
+      longitude: $longitude
+    )
+  }
+`
+
+function DataFunction () {
+  const { loading, error, data } = useQuery(units)
+
+  if (loading) return 'Loading...'
+  if (error) return `Error! ${error.message}`

```

Comment:
You need to adjust the view based on if the query is loading/error or not. Right now you have the assumption in rest of the code that this will return an array, but in the real world, this actually returns a string for a brief period of time(or on an error, indefinitely). So how do we handle those? We need to present that result to the user.